adminapi/dataset.py

```diff
@@ -230,6 +231,32 @@ def get_free_ip_addrs(self):
                     if host not in used_hosts:
                         yield host
 
+    def get_free_ip_addr(self, lock=True):
+        """Get one free IP address from network
+
+        This will give you exactly one free IP address for the queried network
+        and lock it to avoid somebody else using it. If you do not care you
+        can set lock to false but may need to handle the CommitError with the
+        duplicate IP on your own.
+
+        :param lock: Lock free IP address for 60 seconds
+
+        :return:
+        """
+
+        if not lock:
+            return next(self.get_network_ip_addrs())
+
+        lock_api = api.get('api')
+        try:
+            for free_ip in self.get_free_ip_addrs():
+                success = lock_api.lock(free_ip)
+
+                if success is True:
+                    return free_ip
+        except StopIteration:
+            raise AdminapiException('No free IPs left!')
```

Comment:
Why use the try: except: block here?
I think we can remove it:

```
        for free_ip in self.get_free_ip_addrs():
            if lock_api.lock(free_ip) is True:
                return free_ip

        raise AdminapiException('No free IPs left!')
```

Reply:
Meik is right there. Interesting comment on Stackoverflow on that: https://stackoverflow.com/a/14413978
`for` is checking for `StopIteration` and will just stop. So Meik's suggestion is correct.
```suggestion
        or free_ip in self.get_free_ip_addrs():
            if lock_api.lock(free_ip) is True:
                return free_ip

        raise AdminapiException('No free IPs left!')
```
If `lock_api.lock(free_ip)` is also not returning anything else truelike, then the explicit `is True` might also be not necessary:
```suggestion
        for free_ip in self.get_free_ip_addrs():
            if lock_api.lock(free_ip):
                return free_ip

        raise AdminapiException('No free IPs left!')
```

Reply:
`lock_api.lock(free_ip)` returns a number or True so we need to check for exactly `True` here